src/lib/y2partitioner/device_graphs.rb

```diff
@@ -44,16 +44,31 @@ def create_instance(system, initial)
     end
 
     # Devicegraph representing the system
+    #
+    # @return [Y2Storage::Devicegraph]
     attr_reader :system
+
+    # Devicegraph representing the initial state
+    #
+    # @note The initial state is useful to known if something has been modified.
+    #
+    # @return [Y2Storage::Devicegraph]
+    attr_reader :initial
+
     # Working Devicegraph, to be modified during the partitioner execution
+    #
+    # @return [Y2Storage::Devicegraph]
     attr_accessor :current
 
+    # Disk analyzer for the system graph
+    #
+    # @return [Y2Storage::DiskAnalyzer]
     attr_reader :disk_analyzer
 
     def initialize(system: nil, initial: nil)
       @system = system || storage_manager.probed
-      initial ||= storage_manager.staging
-      @current = initial.dup
+      @initial = initial || storage_manager.staging
```

Comment:
you can have it in method itself, as it is evaluated lazy.

try it youtself in easy script:
```
class A
  def random
    rand(5)
  end

  def initialize(a: random)
    puts a
  end
end

A.new
A.new
A.new
A.new
```

Reply:
Umm, I have realized that in this case we don't want only a default value, we want to use `storage` values even when `nil` is passed as value for `:system` and `:initial`.

Reply:
then please document it as it is not much obvious.